// Copyright 2024 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
// All generated protos should be opaque, with "xxx_hidden_" prepended to their field names.


package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";


message NetworksecurityAlphaServerTlsPolicy {
  string name = 1;
  string description = 2;
  string create_time = 3;
  string update_time = 4;
  map<string, string> labels = 5;
  bool allow_open = 6;
  NetworksecurityAlphaServerTlsPolicyServerCertificate server_certificate = 7;
  NetworksecurityAlphaServerTlsPolicyMtlsPolicy mtls_policy = 8;
  string project = 9;
  string location = 10;
}

message NetworksecurityAlphaServerTlsPolicyServerCertificate {
  NetworksecurityAlphaServerTlsPolicyServerCertificateLocalFilepath local_filepath = 1;
  NetworksecurityAlphaServerTlsPolicyServerCertificateGrpcEndpoint grpc_endpoint = 2;
  NetworksecurityAlphaServerTlsPolicyServerCertificateCertificateProviderInstance certificate_provider_instance = 3;
}

message NetworksecurityAlphaServerTlsPolicyServerCertificateLocalFilepath {
  string certificate_path = 1;
  string private_key_path = 2;
}

message NetworksecurityAlphaServerTlsPolicyServerCertificateGrpcEndpoint {
  string target_uri = 1;
}

message NetworksecurityAlphaServerTlsPolicyServerCertificateCertificateProviderInstance {
  string plugin_instance = 1;
}

message NetworksecurityAlphaServerTlsPolicyMtlsPolicy {
  repeated NetworksecurityAlphaServerTlsPolicyMtlsPolicyClientValidationCa client_validation_ca = 1;
}

message NetworksecurityAlphaServerTlsPolicyMtlsPolicyClientValidationCa {
  string ca_cert_path = 1;
  NetworksecurityAlphaServerTlsPolicyMtlsPolicyClientValidationCaGrpcEndpoint grpc_endpoint = 2;
  NetworksecurityAlphaServerTlsPolicyMtlsPolicyClientValidationCaCertificateProviderInstance certificate_provider_instance = 3;
}

message NetworksecurityAlphaServerTlsPolicyMtlsPolicyClientValidationCaGrpcEndpoint {
  string target_uri = 1;
}

message NetworksecurityAlphaServerTlsPolicyMtlsPolicyClientValidationCaCertificateProviderInstance {
  string plugin_instance = 1;
}

message ApplyNetworksecurityAlphaServerTlsPolicyRequest {
  NetworksecurityAlphaServerTlsPolicy resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteNetworksecurityAlphaServerTlsPolicyRequest {
  string service_account_file = 1;
  NetworksecurityAlphaServerTlsPolicy resource = 2;
}

message ListNetworksecurityAlphaServerTlsPolicyRequest {
  string service_account_file = 1;
  string Project = 2;
  string Location = 3;
}

message ListNetworksecurityAlphaServerTlsPolicyResponse {
  repeated NetworksecurityAlphaServerTlsPolicy items = 1;
}

service NetworksecurityAlphaServerTlsPolicyService {
  rpc ApplyNetworksecurityAlphaServerTlsPolicy(ApplyNetworksecurityAlphaServerTlsPolicyRequest) returns (NetworksecurityAlphaServerTlsPolicy);
  rpc DeleteNetworksecurityAlphaServerTlsPolicy(DeleteNetworksecurityAlphaServerTlsPolicyRequest) returns (google.protobuf.Empty);
  rpc ListNetworksecurityAlphaServerTlsPolicy(ListNetworksecurityAlphaServerTlsPolicyRequest) returns (ListNetworksecurityAlphaServerTlsPolicyResponse);
}
